### ISSUE TITLE
Give better control on select tag helper in terms of selected and disabled options

### ISSUE THREAD

**Issue by prikha** [rails/rails#32080]
### Steps to reproduce

Consider the following case - I have a signup form with a select field. The field is required and should contain a placeholder. As we all know we can achieve this by using `select` and `:prompt` option. 

Example from docs: 
```ruby
select("post", "person_id", Person.all.collect {|p| [ p.name, p.id ] }, {prompt: 'Select Person'})
```
outputs 
```html
<select name="post[person_id]" id="post_person_id">
  <option value="">Select Person</option>
  <option value="1">David</option>
  <option value="2">Eileen</option>
  <option value="3">Rafael</option>
</select>
```
An important detail is that placeholder MUST be disabled. This is logically correct. If you have a required field you do not want anyone to submit a placeholder and displaying error messages is also not that fancy.

Going forward one can assume that if we have an option tag with a value equals to an empty string, that means you can employ `:selected` and `:disabled` options to gain precise control over it. It works on any option provided by the client but not on the default one.

### Actual behavior

```ruby
select("post", "person_id", Person.all.collect {|p| [ p.name, p.id ] }, {prompt: 'Select Person', selected: '', disabled: ''})
```

```html
<select name="post[person_id]" id="post_person_id">
  <option value="">Select Person</option>
  <option value="1">David</option>
  <option value="2">Eileen</option>
  <option value="3">Rafael</option>
</select>
```

### Expected behavior
The usage of generic helper methods should be transparent. I expect provided options not to be silently ignored but to be placed as attributes on a placeholder option.

```html
<select name="post[person_id]" id="post_person_id">
  <option value="" disabled="disabled" selected="selected">Select Person</option>
  <option value="1">David</option>
  <option value="2">Eileen</option>
  <option value="3">Rafael</option>
</select>
```

One may notice that it is possible to pass empty option manually:

```ruby
helper.select("post", "person_id", [['Select name', ''], ["Peter", 1], ["Maria", 2], ["Justas", 3]], {required: true, selected: '', disabled: ''})
```
```html
<select name="post[person_id]" id="post_person_id">
<option selected="selected" disabled="disabled" value="">Select name</option>
<option value="1">Peter</option>
<option value="2">Maria</option>
<option value="3">Justas</option></select>
```

well it would be sufficient unless `required` option jumps in and produces another empty option:

```ruby
helper.select("post", "person_id", [['Select name', ''], ["Peter", 1], ["Maria", 2], ["Justas", 3]], {required: true, selected: '', disabled: ''}, {required: true}) 
```

```html
<select required="required" name="post[person_id]" id="post_person_id">
<option value=""></option>
<option selected="selected" disabled="disabled" value="">Select name</option>
<option value="1">Peter</option>
<option value="2">Maria</option>
<option value="3">Justas</option></select>
```

A lot of stuff has been built on top of this implementation([simple_form](https://github.com/plataformatec/simple_form/) is among those, needless to say, that a lot of custom stuff is based on `simple_form` itself. ). And it is strange nobody mentions this, implementing various hacks around it.

And the fix is really straightforward: 

replace this method 

https://github.com/rails/rails/blob/v5.1.4/actionview/lib/action_view/helpers/tags/base.rb#L166-L174

```ruby
        def add_options(option_tags, options, value = nil)
          if options[:include_blank]
            option_tags = tag_builder.content_tag_string("option", options[:include_blank].kind_of?(String) ? options[:include_blank] : nil, value: EMPTY_STRING) + "\n" + option_tags
          end
          if value.blank? && options[:prompt]
            opts = {value: EMPTY_STRING}.tap do |prompt_opts|
              prompt_opts[:disabled] = true if options[:disabled] == EMPTY_STRING
              prompt_opts[:selected] = true if options[:selected] == EMPTY_STRING
            end
            option_tags = tag_builder.content_tag_string("option", prompt_text(options[:prompt]), opts) + "\n" + option_tags
          end
          option_tags
        end
```

I can submit a PR if a proposal would be accepted by the community.

### System configuration
**Rails version**: 5.1.4

**Ruby version**: 2.4.2

**Comment by matthewd** [comment]
> And the fix is really straightforward

If you have a code change to suggest, please do submit a PR: they are specifically designed to make changes easy to read and review.

**Comment by prikha** [comment]
Sure

**Comment by prikha** [comment]
@matthewd added a PR

**Comment by y-yagi** [comment]
Fixed by #32087
